src/TwigComponent/src/Twig/ComponentLexer.php

```
@@ -0,0 +1,152 @@
+<?php
+
+namespace Symfony\UX\TwigComponent\Twig;
+
+use Twig\Lexer;
+use Twig\Source;
+use Twig\TokenStream;
+
+class ComponentLexer extends Lexer
+{
+    public const ATTRIBUTES_REGEX = '(?<attributes>(?:\s+[\w\-:.@]+(=(?:\\\"[^\\\"]*\\\"|\'[^\']*\'|[^\'\\\"=<>]+))?)*\s*)';
+    public const OPEN_TAGS_REGEX = '/<\s*x-(?<name>([[\w\-\:\.]+))\s*'.self::ATTRIBUTES_REGEX.'(\s?)+>/';
```

This feels like it's matching a bit too greedily. For example, I had originally called my component `success_alert` and I think `_` is not an expected character in the name. So, the opening tag failed to match... but then the ending tag DID match, and I got a very cryptic:

> Unexpected "endcomponent" tag

Behind-the-scenes, my code had been half-lexed to this form, which caused the error:

```
    <x-success-alert type="danger" message="another alert!">
        {% block whatever %}sdfsdf{% endblock %}
    {% endcomponent %}
```

Also, I can't get this code to work:

```
<x-successAlert type="danger">
```

It never matches the `OPEN_TAGS_REGEX` unless I remove all of the attributes (i.e. `<x-successAlert>`). Am I doing something wrong or is there a bug?

---

> I think _ is not an expected character in the name 

No it actually is

> Also, I can't get this code to work:

Sorry I know what is it replace your " by '
`<x-successAlert type='danger'>`

I am working on a fix to support argument with " and '

---

If possible, it'd be great to identify a matching opening tag - e.g. `<x-successAlert` FIRST with the regex. And then walk through the rest of the source looking for attributes and eventually the closing `>` or `/>`. That may be beyond the scope of this PR (and Twig internals are COMPLEX). But the motivation is to give better errors. For example, suppose a user does this:

```
<x-successAlert type='danger>
```

I'm not sure what error that would give currently (it might not even match the regex and then would fail silently - which is kind of the worst situation). But ideally, it would be something like:

> Unexpected character `>`: was expecting `'` to close the attribute string.

Thanks :)